R/mobr.R

```diff
@@ -10,221 +10,244 @@ require(pracma)
 ## sample-based implicit, and individual-based rescaled to sample
 get_avg_dens = function(dat_sp, dat_plot, ScaleBy) {
     # Auxillary function to obtain average density within plot for rescaling Ask user
-    # to specify which treatment is used as 'standard' for rescaling.
+    # to specify which group is used as 'standard' for rescaling.
     if (!is.na(ScaleBy)) {
-        dat_scale = dat_sp[match(dat_plot[which(dat_plot[ , 2] == ScaleBy), 1],
-                                 dat_sp[ , 1]), 2:ncol(dat_sp)]
-        avg_dens = mean(apply(dat_scale, 1, sum))
+        avg_dens = mean(apply(dat_sp[dat_plot$group == ScaleBy, ], 1, sum))
```

Comment:
I'd lean towards the former (ie having a function to do the check and maybe throwing a warning if they don't match), just to be safe; but if most ecologists are already used to the way `vegan` works, that's fine too.